hack/update-staging-godeps.sh

```
@@ -0,0 +1,103 @@
+# TODO client-go should probably be made consistent
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+
+# keep the godep restore path reasonably stable to avoid unnecessary restores
+godepRestoreDir=${TMPDIR:-/tmp/}/kube-godep-restore
+
+TARGET_DIR=${TARGET_DIR:-${KUBE_ROOT}/staging}
+echo "working in ${TARGET_DIR}"
+
+ORIGINAL_GOPATH="${GOPATH}"
+
+SKIP_RESTORE=${SKIP_RESTORE:-}
+if [ "${SKIP_RESTORE}" != "true" ]; then
+	echo "starting godep restore"
+	mkdir -p "${godepRestoreDir}"
+
+	# add the vendor folder so that we don't redownload things during restore
+	GOPATH="${godepRestoreDir}:${KUBE_ROOT}/staging:${ORIGINAL_GOPATH}"
+	# restore from kubernetes godeps to ensure we get the correct levels
+	# you get errors about the staging repos not using a known version control system
+	godep restore > ${godepRestoreDir}/godep-restore.log	2>&1 || true
+	echo "finished godep restore"
+fi
+
+echo "forcing fake godep folders to match the current state of master in tmp"
+for stagingRepo in $(ls ${KUBE_ROOT}/staging/src/k8s.io); do
+	echo "    creating ${stagingRepo}"
+	rm -rf ${godepRestoreDir}/src/k8s.io/${stagingRepo}
+	cp -R ${KUBE_ROOT}/staging/src/k8s.io/${stagingRepo} ${godepRestoreDir}/src/k8s.io
+
+	# we need a git commit in the godep folder, otherwise godep won't save
+	pushd ${godepRestoreDir}/src/k8s.io/${stagingRepo}
+	git init > /dev/null
+	# we need this so later commands work, but nothing should ever actually include these commits
+	git config user.email "you@example.com"
+	git config user.name "Your Name"
+	git add . > /dev/null
+	git commit -m "fake commit" > /dev/null
+	popd
+done
+
+function updateGodepManifest() {
+	repo=${1}
+
+	echo "starting ${repo} save"
+	mkdir -p ${TARGET_DIR}/src/k8s.io
+	# if target_dir isn't the same as source dir, you need copy
+	cp -R ${KUBE_ROOT}/staging/src/${repo} ${TARGET_DIR}/src/k8s.io || true
```

`|| true` swallows errors. not nice